### ISSUE TITLE
TaskDone in REST

### ISSUE THREAD

**Issue by nicogranelli** [e2ebridge/bpmn#11]
Probably this is an stupid question, but how do I tell the engine that a task is done using the REST server?

**Comment by nicogranelli** [comment]
Could be sending a trigger event with TaskName + "Done"? 

I will have to install and test. This question came up when reading the docs and thinking if I can use this project, before even downloading. My points is that if I'm correct, it could be something to add to the docs.

**Comment by cyrilschmitt** [comment]
Yes you are right the TaskName+"Done" event is used to end a waiting task and that it could be added to the docs

**Comment by Aditya267** [comment]
How to trigger the MyTask done using rest call.
I have started the process using 
client.post('/Task/MyStart', function(err, req, res, obj) {..}.

I have got the process id and is trying to trigger MyTaskDone, because MyTask is a waiting task.
I am trying this code to trigger the MyTaskDone
client.put('/Task/processID/MyTaskDone/eventID[eventID i don't know where to get it]', data, function(err, req, res, obj) {

But I am not able to do it because i don't know the event id or may be the syntax I m using is wrong.
Can you please help me out

**Comment by cyrilschmitt** [comment]
The eventID is defined by the client and must be unique for the process and event. This is made so if you send the same event twice it won't be triggered twice. That's why we use put because it is idempotent.

You can have a look in function [sendMessage](https://github.com/e2ebridge/bpmn/blob/master/lib/rest.js#LC359) to see the implementation.

**Comment by Aditya267** [comment]
Thanks!!!.I gave some random event ID and its working now

**Comment by dawn5527** [comment]
How to create the process using
client.post('/Task', function(err, req, res, obj) {..}   ?
console.log(err); prints :
{ [InvalidArgumentError: Could not find process name 'Task'.]
  message: 'Could not find process name \'Task\'.',
  statusCode: 409,
  body: 
   { code: 'InvalidArgument',
     message: 'Could not find process name \'Task\'.' },
  restCode: 'InvalidArgument',
  name: 'InvalidArgumentError' }
